ff2mpv.js

```diff
@@ -99,75 +97,96 @@ async function createProfile(profile) {
 
-  const profiles = (await getProfiles())
-    .filter(pf => pf.id !== menuItemId);
+  const profiles = (await getProfiles()).filter((pf) => pf.id !== menuItemId);
 
   if (profiles.length === 0) {
     changeToSingleEntry();
   }
 }
 
 function updateProfile(profile) {
-  browser.contextMenus.update(profile.id, {
+  chrome.contextMenus.update(profile.id, {
     title: profile.name,
   });
 }
 
-browser.browserAction.onClicked.addListener((tab) => {
-  ff2mpv(tab.url);
+chrome.action.onClicked.addListener((tab) => {
+  ff2mpv(tab.url, tab.id);
 });
 
-// Messages sent with browser.runtime.sendMessage (https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/runtime/sendMessage) from external applications will be handle here.
+// Messages sent with chrome.runtime.sendMessage (https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/runtime/sendMessage) from external applications will be handle here.
 // ref: https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/runtime/onMessageExternal
-browser.runtime.onMessageExternal.addListener((request, sender, sendResponse) => {
+chrome.runtime.onMessageExternal.addListener((request, sender, sendResponse) => {
   if (!request) {
-    console.warn('No request in external message');
+    console.warn("No request in external message");
     return;
   }
 
   const { type, url } = request;
-  console.debug('Request from:', sender);
+  console.debug("Request from:", sender);
 
   switch (type) {
     case OPEN_VIDEO:
       ff2mpv(url);
-      return sendResponse('ok');
+      return sendResponse("ok");
     default:
-      console.warn('No handler for external type:', type);
+      console.warn("No handler for external type:", type);
       return;
   }
 });
 
-getOS().then(async (os) => {
-  TITLE = os === "win" ? "Play in MP&V" : "Play in MPV (&W)";
-
+chrome.runtime.onInstalled.addListener(async (_) => {
   const profiles = await getProfiles();
 
   if (profiles.length === 0) {
     changeToSingleEntry();
   } else {
     changeToMultiEntries();
 
-    profiles.forEach(profile => {
-      browser.contextMenus.create({
+    profiles.forEach((profile) => {
+      chrome.contextMenus.create({
         parentId: "ff2mpv",
         id: profile.id,
         title: profile.name,
         contexts,
-        onclick: submenuClicked,
-      })
+      });
     });
   }
 });
+
+chrome.contextMenus.onClicked.addListener(submenuClicked);
+
+chrome.runtime.onMessage.addListener((request, _sender, sendResponse) => {
+  if (!request) {
+    console.warn("No request in message");
+    return;
+  }
+
+  const { type, profile } = request;
+
+  switch (type) {
+    case CREATE_PROFILE:
+      createProfile(profile);
+      break;
+    case UPDATE_PROFILE:
+      updateProfile(profile);
+      break;
+    case DELETE_PROFILE:
+      deleteProfile(profile.id);
+      break;
+    default:
+      console.warn("No handler for type:", type);
+      return;
+  }
```

Comment:
We need to call `sendResponse` or it will log an error because the message didn't get a reply.

It could be just this for now as the options is not using the reply.
```javascript
  sendResponse("ok");
```